File: tests/tooling/perf/add.c
#include <stdio.h>
int main(int argc, char **argv) {
    double acc = 0, a = argc;
    for (int r = 0; r < 20; r++)
        for (int i = 0; i < 1000000; i++) acc += a;
    printf("%.0f\n", acc);
    return 0;
}

File: tests/tooling/perf/add.rs
// The Rust twin of add.fpp: a float accumulation, nothing else. `a` comes
// from argc for the same reason C's does — a constant would fold the loop
// away and the benchmark would measure nothing.
fn main() {
    let a = std::env::args().count() as f64;
    let mut acc: f64 = 0.0;
    for _ in 0..20 {
        for _ in 0..1000000 { acc += a; }
    }
    println!("{:.0}", acc);
}
